p2p/service.go

```diff
@@ -133,36 +133,39 @@ func (s *Service) receiveCoreMessages() {
 		// receive message from core service
 		msg := <-s.msgRec
 
-		// check if non-status message
-		if msg.GetType() != StatusMsgType {
-
-			log.Trace(
-				"Broadcasting message from core service",
-				"host", s.host.id(),
-				"type", msg.GetType(),
-			)
-
-			// broadcast message to connected peers
-			s.host.broadcast(msg)
-
-		} else {
-
-			// check if status is enabled
-			if !s.noStatus {
+		log.Trace(
+			"Broadcasting message from core service",
+			"host", s.host.id(),
+			"type", msg.GetType(),
+		)
 
-				// update host status message
-				s.status.setHostMessage(msg)
-			}
-		}
+		// broadcast message to connected peers
+		s.host.broadcast(msg)
 	}
 }
 
 // handleConn starts processes that manage the connection
 func (s *Service) handleConn(conn network.Conn) {
-
 	// check if status is enabled
 	if !s.noStatus {
 
+		// get latest block header from block state
+		latestBlock := s.cfg.BlockState.LatestHeader()
```

Comment:
would it be preferred to create a field in the service itself for BlockState and StorageState so we don't need to keep the whole config?

Reply:
I feel like I did this at some point and then reverted but I don't remember why. Will check again.

Reply:
So we already need several items from the config (`ProtocolVersion`, `MinSupportedVersion`, `Roles`) in addition to needing `BlockState`. Should I move them all to the service struct?

Reply:
i see, in that case this makes sense